### PR TITLE
Cypress refactoring plus slack

### PR DESCRIPTION
**Pull request by FahadDarw**
### Description

This pull request addresses the following aspects:

1. **Refactoring Cypress Tests:** The Cypress tests in `sponsored-conversation.cy.js` have been refactored to improve readability, maintainability, and overall code quality and more. The changes ensure that the tests are more robust and follow best practices.

2. **Add Slack Notification:** A new feature has been introduced to enhance the test reporting process. Upon test completion, a Slack notification will be sent to the designated channel, providing visibility into the test results.

#### Refactoring Cypress Tests

- Improved code organisation and structure in `sponsored-conversation.cy.js`.
- Updated selectors and assertions for better reliability.
- Addressed any code smells and enhanced overall code quality.

#### Add Slack Notification

- Integrated Slack notification functionality to notify the specified channel upon test completion.
- Configured necessary environment variables for Slack integration.

### Fixes

- Addressed any issues or failures in the previous version of `sponsored-conversation.cy.js`.
  
### Screenshots

image

(1 test is failing due to this ([Bug 151197](https://dfe-gov-uk.visualstudio.com/Academies-and-Free-Schools-SIP/_workitems/edit/151197): Bug Report - Issue with Link Field Validation), which is being fixed by Ash, once it is fixed the test should pass as normal. 

### Checklist

- [x] Cypress tests pass locally & Dev Env
- [x] Added/Updated documentation if necessary.
- [x] Followed coding style and best practices.
- [x] Have been looked at by @cshnimble

### CODE REVIEW

File: .github/workflows/build-and-push-image.yml [hunk 1]
```diff
@@ -173,3 +173,9 @@ jobs:
         with:
           name: screenshots-${{ needs.set-env.outputs.environment }}
           path: Dfe.PrepareConversions/Dfe.PrepareConversions.CypressTests/cypress/screenshots
+          
+      - name: Report results
+        if: always()
+        run: npm run cy:notify -- --custom-text="Environment ${{ needs.set-env.outputs.environment }}, See more information https://github.com/DFE-Digital/prepare-academy-conversions/actions/runs/${{github.run_id}}"
+        env:
+          SLACK_WEBHOOK_URL: ${{ secrets.SLACK_WEBHOOK_URL }}
```
Comment (dneed-nimble) [inline .github/workflows/build-and-push-image.yml:181]: Have you added this secret?

Comment (FahadDarw) [reply .github/workflows/build-and-push-image.yml:181]: Yes it has been added to the repo. as   SLACK_WEBHOOK_UR. Thanks

Comment (dneed-nimble) [reply .github/workflows/build-and-push-image.yml:181]: Has it been added to Terraform etc so it'll be in dev and test?

**Review by dneed-nimble — APPROVED** [review]

File: .github/workflows/build-and-push-image.yml (hunk 1 above)
Comment (FahadDarw) [reply .github/workflows/build-and-push-image.yml:181]: I don't think we need it on Terraform, it just need to be on GitHub.

**Pull request merged**